syscalls_linux.go

```diff
@@ -113,6 +113,18 @@ func setDeviceOptions(fd uintptr, config Config) (err error) {
 	if config.Persist {
 		value = 1
 	}
+
+	if config.NonBlocking {
+		if err := syscall.SetNonblock(int(fd), true); err != nil {
```

Comment:
This could be just `if err = syscall.` as it's already declared. Not a big deal though.